controllers/grafanafolder_controller.go

```diff
@@ -169,21 +169,25 @@ func (r *GrafanaFolderReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
 
-	if folder.Spec.InstanceSelector == nil {
-		controllerLog.Info("no instance selector found for folder, nothing to do", "name", folder.Name, "namespace", folder.Namespace)
-		return ctrl.Result{RequeueAfter: RequeueDelay}, nil
-	}
-
 	instances, err := GetMatchingInstances(ctx, r.Client, folder.Spec.InstanceSelector)
 	if err != nil {
 		controllerLog.Error(err, "could not find matching instances", "name", folder.Name)
+		folder.Status.NoMatchingInstances = true
+		err = r.Client.Status().Update(ctx, folder)
+		if err != nil {
+			controllerLog.Error(err, "unable to update status", "name", folder.Name)
+		}
 		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
-	// your logic here
 
 	if len(instances.Items) == 0 {
 		controllerLog.Info("no matching instances found for folder")
-		return ctrl.Result{Requeue: false}, nil
+		folder.Status.NoMatchingInstances = true
+		err = r.Client.Status().Update(ctx, folder)
+		if err != nil {
+			controllerLog.Error(err, "unable to update status", "name", folder.Name)
+		}
+		return ctrl.Result{RequeueAfter: RequeueDelay}, err
 	}
 
```

Comment:
Should we add `folder.Status.NoMatchingInstances: false` somewhere around this line?

Reply:
I remove it instead.